cot/src/handler.rs

```diff
@@ -70,11 +70,12 @@ pub(crate) fn into_box_request_handler<T, H: RequestHandler<T> + Send + Sync>(
 
 macro_rules! impl_request_handler {
     ($($ty:ident),*) => {
-        impl<T, $($ty,)* R> RequestHandler<($($ty,)*)> for T
+        impl<Func, $($ty,)* Fut, R> RequestHandler<($($ty,)*)> for Func
         where
-            T: FnOnce($($ty,)*) -> R + Clone + Send + Sync + 'static,
+            Func: FnOnce($($ty,)*) -> Fut + Clone + Send + Sync + 'static,
             $($ty: FromRequestParts + Send,)*
-            R: for<'a> Future<Output = Result<Response>> + Send,
+            Fut: Future<Output = Result<R>> + Send,
```

Comment:
I'm wondering, do we actually need the return type to be wrapped in `Result`? Any chance we could just change this into the following:

```suggestion
            Fut: Future<Output = R> + Send,
```

? Sounds like a sensible thing to do given `IntoResponse` also seems to be implemented for `Result<T, E>`.

Reply:
We absolutely can change it to plain `R`; however, if we do that, we'll lose the ability to handle user errors, as those would be transformed into `Response`. I don't consider this a problem, I actually think it's the good solution, I just want to make sure you're aware of this. Basically, this:

https://github.com/cot-rs/cot/blob/46d6ba0fd8c04fe671af1f7415988834f0b64083/cot/src/handler.rs#L89-L92

would become this:
```rust
let response = self.clone()($($ty,)*).await;

Ok(response)
```

Reply:
> I don't consider this a problem

It definitely **is** a problem; this is one of the most distinct features of our framework, and I don't think there's any reason to throw this away just like that. We have ways to handle user errors, and I don't think there's a need to add another one that's incompatible with what we already have.

Let me rephrase the question: why doesn't `IntoResponse::into_response()` return `cot::Result<Response>`?

Reply:
> Let me rephrase the question: why doesn't `IntoResponse::into_response()` return `cot::Result<Response>`?

Because I didn't think of it :sweat_smile: That's actually a great idea, let me implement that!